File: src/hebbs/services/health.py
"""Async wrapper for the HEBBS HealthService gRPC method."""

from __future__ import annotations

from hebbs._generated import hebbs_pb2, hebbs_pb2_grpc
from hebbs.exceptions import _map_grpc_error
from hebbs.types import HealthStatus


class HealthServiceClient:
    """Async client for the HEBBS HealthService."""

    def __init__(self, stub: hebbs_pb2_grpc.HealthServiceStub) -> None:
        self._stub = stub

    async def check(self) -> HealthStatus:
        try:
            resp = await self._stub.Check(hebbs_pb2.HealthCheckRequest())
        except Exception as e:
            raise _map_grpc_error(e) from e

        return HealthStatus(
            serving=(resp.status == hebbs_pb2.HealthCheckResponse.SERVING),
            version=resp.version,
            memory_count=resp.memory_count,
            uptime_seconds=resp.uptime_seconds,
        )

File: src/services/health.ts
/**
 * Async wrapper for the HEBBS HealthService gRPC method.
 */

import type { Metadata } from '@grpc/grpc-js';
import { mapGrpcError } from '../errors.js';
import { grpcUnary, SERVING_STATUS_SERVING } from '../proto.js';
import type { HealthStatus } from '../types.js';

/* eslint-disable @typescript-eslint/no-explicit-any */

export class HealthService {
  constructor(
    private readonly stub: any,
    private readonly metadata: Metadata,
  ) {}

  async check(): Promise<HealthStatus> {
    try {
      const resp = await grpcUnary<any>((cb) =>
        this.stub.check({}, this.metadata, cb),
      );
      return {
        serving: resp.status === SERVING_STATUS_SERVING,
        version: resp.version ?? '',
        memoryCount: resp.memoryCount ?? resp.memory_count ?? 0,
        uptimeSeconds: resp.uptimeSeconds ?? resp.uptime_seconds ?? 0,
      };
    } catch (e) {
      throw mapGrpcError(e);
    }
  }
}

/* eslint-enable @typescript-eslint/no-explicit-any */
